### DISCUSSION TITLE
Misleading information regarding "ProcessRequestBody()"

### DISCUSSION THREAD

**Discussion by noboruma** [corazawaf/coraza#1063] [Q&A]
The documentation around this function says:
```
	// ProcessRequestBody Performs the analysis of the request body (if any)
	//
	// This method perform the analysis on the request body. It is optional to
	// call that function. If this API consumer already knows that there isn't a
	// body for inspect it is recommended to skip this step.
	//
	// Remember to check for a possible intervention.
```
Implying that if you don't have any body, there is no need to call this function.
Unless I am missing another way of achieving this, there is currently no other way to go beyond phase 1 without calling this function?
Meaning that even if there is no body, this function should be called to process phase 2 rules that would match headers for instance.

**Comment by jcchavezs** [comment]
Yeah, sounds misleading to me for the reasons you mention. What do you
think @m4tteoP?

**Reply by M4tteoP** [reply]
Thanks for reporting it, that's interesting and I agree that sounds misleading, especially taking into account CRS rules where quite a lot of headers and args are checked with phase 2 rules. At first sight, I think we should rewrite it.

I recall that also libmodsecurity comes with [this description](https://github.com/owasp-modsecurity/ModSecurity/blob/c8056483f7dbf5fc267962324aeffe5ff60f3467/src/transaction.cc#L769-L774), so I dropped a line in the [modsec slack](https://owasp.slack.com/archives/C069PCXSW12/p1715463196780949) also to have their opinion.
